packages/hooks/use-timeout/index.ts

```diff
@@ -0,0 +1,20 @@
+import { onBeforeUnmount } from 'vue'
+
+export default function () {
```

Comment:
I would suggest something more like react like.

```
const cancel = useTimeout(fn, delayInMs)
/// use cancel in other code
```

Reply:
But how do we re-use the handle then? Since each `setup` only gets called once.

Reply:
But consider usage like this:

```
function timeoutInTimeout() {
    useTimeout(() => anotherTimeoutTask（）， 500)
}
useTimeout(timeoutInTimeout, 500)
```

Reply:
@JeremyWuuuuu we don't reuse timeout. Every new useTimeout call create a new timer.
But I think the real problem is for timeout invoked in another async callback, like another timeout.

Reply:
Totally agree, this hook I proposed is meant for some specific event like: `show/hide` for an element, which is mutually exclusive.

Reply:
Maybe we could rename this file to make it more functionality-related so that we won't be confused by the general one though.

Reply:
Since the difference of hook is way different between `React` and `Vue`, I think we can make two of `use-timeout` in order to differentiate them.